### PR TITLE
Add remove message feature

### PR DESCRIPTION
**Pull request by misstick**
### When a message is deleted from `messageComponent`:
 - if conversation still exist (conversation has other messages): 
  - if next message exist: select it,
  - if not select previous message,
  - in both cases: update `:messageID` into`URL`.
 - if not, then conversation is deleted and:
  - `messageListComponent` is updated,
  - select next conversation (upper in the list),
  - update `:conversation:messageID` into URL,

### How is defined nearest message?
 - get next message from conversation,
 - otherwise get previous message from conversation,
 - if message was the last of the conversation: select next conversation into `messageListComponent`,
 - if no nearest message was found then close `conversationComponent`.
 - if page is refreshed by browser by external behaviour (such as browser refresh button or websocket event), nearest message should be found too.

### What is behind `messagesListCounter`?
  - it is the length messages into conversation which takes into account `currentFilter`.
  - if a conversation is displayed into `unseenMailbox`, the counter is only about unread messages from the conversation.

### Deleting a message from `unreadMailbox`
 - [x]  `unreadCounter` should be updated into `mailboxMenu`,
 - if conversation have still unread messages:
  - [x]  update `unreadCounter` into `messageListItem` of mailbox,
  - [x]  remove message from `conversationComponent`,
  - [x]  select the `nearestMessage`,
  - [x]  update URL with `nearestMessageID`.
 - otherwhise:
  - [x] remove message from `messageList`,
  - [x]  select the `nearestMessage`,
  - [x]  update URL with `nearestConversationID/IDnearestMessageID`.
 

### Deleting a message from `trashBox` cant happen
  - [x] `removeButton` must be hidden in this case.

## tasks
 - [x] add `toolbarMessageComponent` with the only button `deleteMessage`,
 - [x] this toolbox only appear for `noTrash` mailbox,
 - make `deleteMessageFeature` work:
  - [x] delete message into `messageStore`,
  - [x] update message list into `routerStore`,
  - [x] remove `messageItem` from `messageListComponent`,
  - [x] remove `messageItem` from `conversationComponent`,
  - if `conversationComponent` is empty:
    - [x] remove conversation from `messageStore`,
    - [x] remove conversation from `messageListComponent`
    - [x] selected nearest conversation,
    - [x] otherwise close conversation,
    - [x] update counter into concerned components: `messageLisItemComponent` or/and `mailboxItemCounterComponent`.

## issues
 - [x] `getMoreMessages` button didnt work on `trashbox`,
 - [x] `displayMessages` wasnt working anymore,
 - [x] redirect to `nearestConversationMessage` if message was removed from `mailbox`,
 - [x] redirect to `messagelist` if conversation was removed from `mailbox`.

### CODE REVIEW

**Comment by misstick** [comment]
lasat coommit is not finished yet; getting nearest message is not working!

**Comment by frankrousseau** [comment]
I propose that we merge that one. It's important that @poupotte start to test it before the end of the milestone.

**Comment by m4dz** [comment]
@misstick the PR is still marked as _in progress_ and I need to review it. Your thoughs?

**Comment by misstick** [comment]
@m4dz: you can review it; 

@frankrousseau : Yes, it was planned :)

**Comment by misstick** [comment]
@m4dz: I'm waiting for your review

File: client/app/actions/message_action_creator.coffee [hunk 1]
```diff
@@ -22,10 +20,11 @@ MessageActionCreator =
             value: message
 
 
-    displayImages: (props={})->
+    displayImages: ({messageID, displayImages=true})->
+        message = MessageStore.getByID messageID
         AppDispatcher.dispatch
-            type: ActionTypes.SETTINGS_UPDATE_REQUEST
-            value: props
+            type: ActionTypes.SETTINGS_UPDATE_RESQUEST
```
Comment (m4dz) [inline client/app/actions/message_action_creator.coffee]: I have fixed the typo in the `SETTINGS_UPDATE_REQUEST` name. You can safely use it here :)

File: client/app/components/conversation.coffee [hunk 2]
```diff
@@ -42,29 +33,35 @@ _freezeHeader = ->
...
+        unless @props.isConversationLoading
+            _freezeHeader.call @
+            _scrollToActive.call @
+
+
+    componentWillReceiveProps: (nextProps) ->
+        # If this conversation was removed
+        # redirect to messages list from mailbox
+        if nextProps.isConversationLoading and not nextProps.messages.length
+            setTimeout ->
```
Comment (m4dz) [inline client/app/components/conversation.coffee]: Just a note: I suppose setTimeout inside view are probably a dark pattern, and we may find amother way to handle let. Just a reminder, let's discuss it later ;)

**Comment by m4dz** [comment]
Thanks @misstick ! Huge and smart job :tada: (and btw, thanks too for cleaning my obviously forgiven dead code :confused:).

**Comment by m4dz** [comment]
woops, just seeing that the PR was reflagged as _in progress_ and I though it was at _to merge_ :disappointed_relieved:. @misstick, tell me if we need to fix it or revert it (should remind that it _just_ test issues that block the PR). So sorry…

**Pull request merged**